mirrord/layer/src/error.rs

```diff
@@ -264,7 +265,32 @@ impl From<HookError> for i64 {
                     _ => libc::EINVAL,
                 },
                 ResponseError::DnsLookup(dns_fail) => match dns_fail.kind {
-                    mirrord_protocol::ResolveErrorKindInternal::Timeout => libc::EAI_AGAIN,
+                    ResolveErrorKindInternal::Timeout => libc::EAI_AGAIN,
+                    ResolveErrorKindInternal::NoConnections => libc::EAI_AGAIN,
+                    ResolveErrorKindInternal::Message(msg) => {
+                        error!("DNS lookup failed with message: {}", msg);
```

Comment:
Setting libc errno isn't the right way to do it. `getaddrinfo` returns the `EAI_*` error as return value directly, not via `errno`.

Reply:
Also, thinking about it - we probably can be lossy until we change the protocol and add it as todo for when we break the protocol.

Reply:
i assumed we would need to set the errno properly as well since we already do:
```
ResponseError::DnsLookup(dns_fail) => match dns_fail.kind {
                    mirrord_protocol::ResolveErrorKindInternal::Timeout => libc::EAI_AGAIN,
                    _ => libc::EAI_FAIL,
                },
```